### ISSUE TITLE
Confusing Multiplication

### ISSUE THREAD

**Issue by jessebett** [malmaud/TensorFlow.jl#336]
I'm having a bit of confusion with the multiplication and was hoping for some insight into what's going on.

Basically, I'm trying to multiply 2 floats and a specific index from a tf.Variable.

```
# Confusing Multiplication
a = [0.01] 
b = [0.02]
x = Variable([1.,2.])
x[1] * a    # > Tensorflow error: Status: In[0] is not a matrix (makes sense, * is for matmul)
x[1] .* a   #> Float64[1] 0.0100  (okay, good!)
x[1].* b    #> Float64[1] 0.0200 (still making sense)
x[1] .* a .* b  #> Tensorflow error: Status: In[0] is not a matrix
(x[1] .* a) .* b #> Tensorflow error: Status: In[0] is not a matrix
x[1] .* (a .* b)  #> Tensorflow error: Status: In[0] is not a matrix

multiply(x[1] .* a , b) #> Float64[1]  0.000200 (This works though)
```

I'm confused why my the cases where I use .* aren't working.

**Comment by oxinabox** [comment]
This is because the changes to the way broadcasting works in julia 0.6 ( see https://julialang.org/blog/2017/01/moredots),
removed our ability to properly overload "dotted" operaters, such as `.*`, to do new functionality, different from there base forms (eg `*`).

Luckily for you I had already started a PR to fix this, before you rose the issue,
and it is now done in  #334.
Unluckily for you, it is dependent on another PR in another package,
so can't just use it now.

I have added your code as a testcase  in #334 and can confirm it is fixed there.

The other work around is to assign things to to intermediary variables.
That will stop fusing.
So long as only one dotted operation is used per expression then fusing does not occur.
